src/mem/alloc.h

```diff
@@ -46,8 +46,10 @@ namespace snmalloc
 // Use flat map is under a single node.
 #  define SNMALLOC_MAX_FLATPAGEMAP_SIZE PAGEMAP_NODE_SIZE
 #endif
-  static constexpr bool USE_FLATPAGEMAP = SNMALLOC_MAX_FLATPAGEMAP_SIZE >=
-    sizeof(FlatPagemap<SUPERSLAB_BITS, uint8_t>);
+  static constexpr bool USE_FLATPAGEMAP =
+    (Pal::pal_features & PalFeatures::LazyCommit) ||
```

Comment:
`largealloc.h` has a typesafe version of this test (`pal_supports()`).  It would be good to reuse that if possible, moving it somewhere more useful if necessary.